src/prefect/infrastructure/docker.py

```diff
@@ -119,14 +127,24 @@ class DockerRegistry(BaseDockerLogin):
 
     @sync_compatible
     async def login(self) -> "DockerClient":
-        return await run_sync_in_worker_thread(
+        warnings.warn(
+            "`login` is deprecated. Instead, use `get_docker_client` to obtain an authenticated `DockerClient`.",
+            category=DeprecationWarning,
+        )
```

Comment:
Usually you need to set a stack level for the warning to display the line where the user made the call. I'd verify this suggests the correct line and adjust accordingly.

```suggestion
            category=DeprecationWarning,
            stacklevel=3
        )
```